var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -92,8 +93,16 @@ def setup_build_environment(self, env):
         env.set("AR", ar.path)
 
         # Manually inject the path of openssl's certs for build.
-        certs = join_path(self.spec["openssl"].prefix, "etc/openssl/cert.pem")
-        env.set("CARGO_HTTP_CAINFO", certs)
+        certs = None
+        for p in ("etc/openssl/cert.pem", "../etc/openssl/cert.pem",
+                  "etc/ssl/certs/ca-bundle.crt", "../etc/ssl/certs/ca-bundle.crt"):
+            certs = join_path(self.spec["openssl"].prefix, p)
+            if os.path.exists(certs):
+                break
+            else:
+                certs = None
+        if certs is not None:
+            env.set("CARGO_HTTP_CAINFO", certs)
```

Comment:
This can also be fixed by building with `rust ^openssl certs=system`. I'd argue we should fix this by instead making system the default variant for `openssl`. @tgamblin @alalazo and I have discussed this before.

Reply:
I can try that, but I'm not sure it can.  The rust-bootstrap package downloads a pre-built rust, which likely has a pre-built OpenSSL in it. Will try it.

Reply:
Yeah, no dice, just like building with the system external.  should probably be using the logic from the OpenSSL package to find the system certs though.

Reply:
Turns out there may be some more meaningful problems here, getting a failure on gitlab using `openssl cert=system` https://gitlab.spack.io/spack/spack/-/jobs/9040548